engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +890,16 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  override def callCacheDataByExecution(id: WorkflowId)(implicit ec: ExecutionContext): Future[Traversable[ExecutionWithCacheData]] = {
+    val action = dataAccess.executionsWithCacheHitWorkflowAndCall(id.toString).result
+    runTransaction(action).map(_.map {
+      case (execution, Some(cacheHitWorkflowUuid), Some(cacheHitCallFqn)) =>
+        ExecutionWithCacheData(execution, Option(CallCacheHit(cacheHitWorkflowUuid, cacheHitCallFqn)))
+      case (execution, None, None) =>
+        ExecutionWithCacheData(execution, None)
+      case (execution, cacheHitWorkflowUuid, cacheHitCallFqn) =>
+        throw new IllegalStateException(s"Expecting no cache hit or (workflow UUID, call FQN) tuple.  Got ($cacheHitWorkflowUuid, $cacheHitCallFqn)")
```

Comment:
Anybody know a better way to handle this case?  Will this turn into a `Future.failure`?